src/NuGet.Services.Metadata.Catalog.Monitoring/Validation/Test/RegistrationValidators.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Protocol;
+using System.Threading.Tasks;
+using NuGet.Protocol.Core.Types;
+using System.Collections.Generic;
+using Microsoft.Extensions.Logging;
+
+namespace NuGet.Services.Metadata.Catalog.Monitoring.Validation.Test
+{
+    public class RegistrationValidators
+    {
+        public static string GetContainsString(bool contains)
+        {
+            return contains ? "contains" : "doesn't contain";
+        }
+        
+        /// <summary>
+        /// Verifies that the index.json blob for the package exists when it should.
+        /// </summary>
+        public class RegistrationIndexExistsValidator : Validator, IValidator<RegistrationEndpoint>
```

Comment:
Not necessarily...there's nothing stopping somebody from writing a validator that depends on two endpoints. I could write a test that compared the Registration and FlatContainer against each other and the V2 feed.

I have, however, created `Validator<T>` which extends `Validator` and `IValidator<T>` for ease of use.